src/structures/Manager.ts

```diff
@@ -190,128 +179,101 @@ export class Manager extends EventEmitter {
-            return resolve(SearchResult);
-        });
+    public async search(searchQuery: SearchQuery, requestor: Snowflake): Promise<SearchResult> {
+        const response = await this.node
+            .makeRequest("GET", `api/tracks/search?query=${encodeURIComponent(searchQuery.query)}&identifier=${searchQuery.identifier || 'ytsearch'}`,)
+            .then(res => res.json());
+
+        if (!response || !response.results) throw new ChiroError("Responded results from the server seems to be empty.");
+        return resolveTracks(response, requestor)
     }
 
     /**
-     * Creates a player instance and add it to players collection
-     * @param {PlayerOptions} options Player Options
-     * @return {Player}
-     *
+     * Creates a player instance and add it to players collection.
+     * 
+     * @param {PlayerOptions} options Player Options to create one, if there is no existing one.
+     * @returns {Player}
      */
     public create(options: PlayerOptions): Player {
-        if (this.players.has(options.guild)) {
-            return this.players.get(options.guild);
-        }
-        const player = new Player(options);
+        let player = this.players.get(options.guild);
+        if (player) return player;
+        player = new Player(options);
         this.emit("playerCreate", player);
         return player;
     }
 
     /**
-     * Send Player
-     * @param {Snowflake} guild Id of Guild
-     * @return {Player}
+     * Get a player by its guild id.
+     * 
+     * @param {Snowflake} guild ID of Guild.
+     * @returns {Player}
      */
     public get(guild: Snowflake): Player | undefined {
         return this.players.get(guild);
     }
+
     /**
-     * Destroy the Node connection
+     * Destroy the Node connection.
      */
-    public destroyNode(): void {
+    public destroyNode() {
         this.node.destroy();
     }
 
     /**
-     * Send Voice State Payload Received from Discord API to Nexus
-     * @param {Object} data
+     * Send Voice State Payload Received from Discord API to Nexus.
+     * 
+     * @param {Object} data The data from the event.
      * @example
-     * client.on('raw', (d)=>{
-     *    manager.updateVoiceState(d);
-     * });
+     * client.on('raw', manager.updateVoiceState);
      */
-    public updateVoiceState(data: any): void {
-        if (
-            !data ||
-            !["VOICE_SERVER_UPDATE", "VOICE_STATE_UPDATE"].includes(
-                data.t || ""
-            )
-        )
-            return;
-        if (data.t === "VOICE_SERVER_UPDATE") {
-            this.node.socket.send(JSON.stringify(data));
-        }
-        if (data.t === "VOICE_STATE_UPDATE") {
-            this.node.socket.send(JSON.stringify(data));
-        }
+    public updateVoiceState(data: any) {
+        if (data && data.t === "VOICE_SERVER_UPDATE" || data.t === "VOICE_STATE_UPDATE") this.node.socket.send(JSON.stringify(data));
```

Comment:
```diff
if (["VOICE_SERVER_UPDATE", "VOICE_STATE_UPDATE"].includes(data?.t)) this.node.socket.send(JSON.stringify(data));
```

Reply:
Well both does the same right? And `includes` means it iterates over the array. Is not using logical operators good? (Just a thought)

Reply:
imo using includes is a lot shorter and cleaner than chaining multiple statements?